addons/crm/models/crm_lead.py

```diff
@@ -430,15 +430,14 @@ def write(self, vals):
 
         return write_result
 
-    def _write_probability(self, vals):
-        compute = False
-        fields_to_check = ['tag_ids', 'stage_id', 'team_id'] + self._pls_get_safe_fields()
-        for field, value in vals.items():
-            if field in fields_to_check:
-                compute = True
-                break
-
-        if not compute:
+    def _write_probability(self, vals, force_compute=False):
+        if not force_compute:
```

Comment:
Maybe this would be clearer?

```
if not self._should_compute_probabilities(vals, force_compute):
    return

...

def _should_compute_probabilities(self, vals, force_recompute):
    if force_recompute:
        return True
    fields_to_check = ['tag_ids', 'stage_id', 'team_id'] + self._pls_get_safe_fields()
    for field, value in vals.items():
        if field in fields_to_check:
            return True
    return False
```

Not sure, I let you decide.

Reply:
after reflexion: this is a bit longer but this will be less confusing (with the force compute) and more readable at the end.
I vote to do what @awa-odoo proposes.

Reply:
Made the changes.